spec/spec_helper.rb

```diff
@@ -67,6 +67,21 @@
   end
 }
 
+add_custom_fact :service_provider, ->(os, facts) {
```

Comment:
What's this for? Can't it just use the actual service_provider fact?

Reply:
The library for rspec-puppet-facts pulls facts from facterdb gem and that gem only includes core Facter facts, not facts from stdlib.  This fact is from stdlib module so the only way to incorporate it into rspec-puppet-facts is to add it using `add_custom_fact`.  This is not using real facter facts during unit tests, that's not possible so all facts are mocked using rspec-puppet-facts to produce real-looking facts during rspec tests.

Reply:
Could you please add a comment to reflect that.

Reply:
comment added